dashboard/app/models/studio_person.rb

```diff
@@ -8,77 +8,115 @@
-    # address. By using hashed_email, we handle the many teachers missing a
-    # plaintext email.
-    hashed_email_to_add = User.hash_email(email_to_add)
-    User.with_deleted.
-      where(user_type: User::TYPE_TEACHER).
-      where(hashed_email: hashed_email_to_add).
-      each do |matching_user|
-      matching_studio_person = matching_user.studio_person
-      begin
-        matching_user.update!(studio_person_id: id)
-      rescue
-        matching_user.update!(studio_person_id: id, email: email_to_add)
-      end
+    users_a = User.where(studio_person_id: studio_person_a.id).pluck(:id)
+    users_b = User.where(studio_person_id: studio_person_b.id).pluck(:id)
 
-      # Merge the matching studio_person (also all associated users), if any,
-      # with this studio_person.
-      if matching_studio_person && self != matching_studio_person
-        matching_studio_person.emails_as_array.each do |existing_email|
-          add_email_to_emails(existing_email)
-        end
-        User.with_deleted.
-          where(studio_person_id: matching_studio_person.id).
-          each do |user|
-          # Bypass validations, as the user (teacher) may be missing an email.
-          user.update_column(:studio_person_id, id)
-          add_email_to_emails(user.email)
-        end
+    # Arbitrarily, we merge the two StudioPersons into studio_person_a.
+    merged_emails_array = (studio_person_a.emails_as_array + studio_person_b.emails_as_array).uniq
+    studio_person_a.update!(emails: merged_emails_array.join(','))
+    unless users_b.empty?
+      User.where(studio_person_id: studio_person_b.id).each do |user|
+        user.update!(studio_person_id: studio_person_a.id)
       end
-
-      # Delete the (now) orphaned studio_person.
-      matching_studio_person.delete if matching_studio_person
     end
 
-    save!
-  end
+    FirehoseClient.instance.put_record(
+      'analysis-events',
+      {
+        study: 'studio_person_audit',
+        event: 'studio_person_merge',
+        data_json: {
+          studio_person_a_id: studio_person_a.id,
+          studio_person_b_id: studio_person_b.id,
+          studio_person_merged_id: studio_person_a.id,
+          user_a_id: users_a,
+          user_b_id: users_b
+        }.to_json
+      }
+    )
 
-  # @returns [Array[String]] An array of emails associated with the studio_person.
-  def emails_as_array
-    return [] if emails.nil?
-    emails.split(',')
+    # Delete the now orphaned StudioPerson.
+    studio_person_b.destroy!
   end
 
-  private
+  # Splits a StudioPerson into multiple StudioPersons, logging the event to Firehose.
+  # @raise [ArgumentError] If the StudioPerson does not have two associated emails, if more than two
+  #   people share the StudioPerson.
+  def self.split(studio_person)
+    unless studio_person.emails_as_array.count == 2
+      raise ArgumentError.new("StudioPerson does not have exactly two emails")
```

Comment:
we can't we split apart more than two users?

Reply:
It isn't so that much we can't, rather the intent is more complicated. For example, if a `StudioPerson` has exactly two emails with two users, it is clear how to split them. If a `StudioPerson` has four emails and three users, how does one go about splitting them automatically? This UI is intended to expedite common tasks, not to solve all possible situations. This is definitely an edge case it does not solve.

Reply:
Ahh, fair enough.